lib/matplotlib/axes/_axes.py

```diff
@@ -2045,14 +2043,16 @@ def make_iterable(x):
             nbars = len(bottom)
             if len(left) == 1:
                 left *= nbars
-            if len(height) == 1:
-                height *= nbars
 
             tick_label_axis = self.yaxis
             tick_label_position = bottom
         else:
             raise ValueError('invalid orientation: %s' % orientation)
 
+        if len(height) == 1:
+            height *= nbars
```

Comment:
It looks like this is leaving in place an old bug--probably harmless, since no one seems to have tripped over it.  The potential problem is that if a kwarg like 'height' is passed in as an ndarray of length 1, it will arrive here and have its value multiplied by nbars, which might not be 1.

`_make_iterable` could be modified to ensure that if an argument comes in with length 1, it is returned as a list; or it could be turned into a `_to_list`, and turn any argument into a list.  The point is that in at least parts of the code the returned object is assumed to be a list, not just an iterable.

I think the `_to_list` approach is the way to go; for these variables, there is no advantage in leaving their type as an indeterminate 'iterable'.